### DISCUSSION TITLE
Why do toast tips always pop up while running QQ?

### DISCUSSION THREAD

**Discussion by IMXeno123** [MaterialCleaner/MaterialCleaner#72] [Q&A] [answered]
Why do toast tips always pop up while running QQ? That's so annoying.

**Comment by IMXeno123** [comment]
The problem has been solved.

**Comment by GuhDoy** [comment]
Thank you for your question.

I understand you're annoyed by the [storage redirect prompts](https://github.com/MaterialCleaner/MaterialCleaner/wiki/Start-To-Use-Storage-Redirect#optimization-for-media-files). For your information, there is only one reason for the storage redirect prompts to appear--the app is interacting with the media store, and the storage redirect may have gotten in the way.

In general cases, storage redirect prompts are always helpful, as their appearance indicates that there may be errors in your mount rules. However, if the obstruction is exactly what you want, it's better to use [Media Provider Manager (MPM for short)](https://github.com/MaterialCleaner/Media-Provider-Manager) because it's not possible to do this with Cleaner at all. For one example, Cleaner can intercept some apps to access media files, but the apps still know that there are media files there (they just cannot be accessed), but MPM can make the app not know that there are media files there. For another example, apps can also obtain files via the media store. You can try to write any mount rules, but it is impossible to prevent MPM from accessing media files on your device.

In the specific case you mentioned, QQ is an app I know to abuse the media store.

**Reply by IMXeno123** [reply, marked as answer]
Thank you for your reply, this issue has been solved. This issue is caused by some rule errors that QQ cannot generate the "./Documents/ringtones/" directory.
